src/Shared/PackageManagers/CondaProjectManager.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.PackageManagers
+{
+    using Contracts;
+    using Microsoft.CST.OpenSource.Helpers;
+    using Microsoft.CST.OpenSource.Model;
+    using Microsoft.CST.OpenSource.PackageActions;
+    using PackageUrl;
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+
+    public class CondaProjectManager : TypedManager<IManagerPackageVersionMetadata, CondaProjectManager.CondaArtifactType>
+    {
+        /// <summary>
+        /// The type of the project manager from the package-url type specifications.
+        /// </summary>
+        /// <seealso href="https://www.github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst"/>
+        public const string Type = "conda";
+        public const string DEFAULT_CONDA_ENDPOINT = "https://repo.anaconda.com/pkgs";
+
+        public override string ManagerType => Type;
+        
+        private const string TarBz2FileType = "tar.bz2";
+        private const string CondaFileType = "conda";
+        private const string BuildQualifier = "build";
+        private const string SubdirQualifier = "subdir";
+        private const string TypeQualifier = "type";
+        private const string DefaultChannel = "main";
+
+        public CondaProjectManager(
+            string directory,
+            IManagerPackageActions<IManagerPackageVersionMetadata>? actions = null,
+            IHttpClientFactory? httpClientFactory = null)
+            : base(actions ?? new NoOpPackageActions(), httpClientFactory ?? new DefaultHttpClientFactory(), directory)
+        {
+        }
+
+        /// <inheritdoc />
+        public override async IAsyncEnumerable<ArtifactUri<CondaArtifactType>> GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)
+        {
+            Check.NotNull(nameof(purl), purl);
+            if (!IsValidPackageUrlForConda(purl))
+            {
+                throw new ArgumentException("Invalid package URL for Conda.");
+            }
+
+            string? type = purl?.Qualifiers?.GetValueOrDefault(TypeQualifier);
```

Comment:
I know that the file type is a part of the package url spec for Conda.  I am wondering about perhaps returning all relevant download urls, and then allowing the caller to filter to which file type they want.  Is it generally expected that only one of the conda or the tar file is generally available, or are both of them usually present?

Reply:
From my analysis, tar.bz2 is more prevalent, in particular in the `free` channel. When the type is not provided, both values are returned (line 54 and 56 below)